src/cli/commands/run.js

```diff
@@ -11,9 +11,22 @@ import map from '../../util/map.js';
 const leven = require('leven');
 const path = require('path');
 
-// Copied from https://github.com/npm/npm/blob/63f153c743f9354376bfb9dad42bd028a320fd1f/lib/run-script.js#L175
+function quoteForCmd(arg: string): string {
+  // See the below blog post for an explanation of what's going on here:
+  // eslint-disable-next-line max-len
+  // https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
```

Comment:
Hah! I'm glad there is another one who read the same article 🗡 

I think you should release this as a library so `yarn` and others can use it from there because I've yet to see a proper Windows-compatible shell escaping library on NPM.